File: Leetcode/Golang/reverseInteger.go
func reverse(x int) int {
    rev := 0
    for x!=0 {
        ld := x%10
        x /= 10
        rev = (rev*10)+ld
    }
     if rev > math.MaxInt32 || rev < math.MinInt32 {
            return 0
        }
    return rev
}

File: Leetcode/java/reverseInteger.java
class reverseInteger {
    public int reverse(int x) {
   
        int rev=0;
        int ld;
        
        while(x!=0)
        {
            ld = x % 10;
            x /= 10;
            if(rev>Integer.MAX_VALUE/10 || rev ==Integer.MAX_VALUE/10 && ld>7)
                return 0;
            if(rev<Integer.MIN_VALUE/10 || rev==Integer.MIN_VALUE/10 && ld<-8)
                return 0;
            rev = (rev *10) + ld;
            
        }
            return rev;
   
    
      
    }
    
}
